drivers/NRF5/Radio_ESB.cpp

```diff
@@ -0,0 +1,664 @@
+	rssi_rx = 0-buffer->rssi;
+
+	// Debug message
+#ifdef MY_DEBUG_VERBOSE_NRF5_ESB
+	NRF5_RADIO_DEBUG(PSTR("NRF5:RX:LEN=%d,NOACK=%d,PID=%d,RSSI=%d,RX=%d\n"),
+	                 buffer->len, buffer->noack, buffer->pid, rssi_rx, buffer->rxmatch);
+#endif
+
+	// release buffer
+	rx_circular_buffer.popBack();
+
+	// Check if radio was disabled by buffer end
+	if (radio_disabled == true) {
+		// Wait until disabling radio is finised
+		while (NRF_RADIO->EVENTS_DISABLED == 0)
+			;
+		// Enable radio
+		NRF5_ESB_startListening();
+	}
+
+	return ret;
+}
+
+void NRF5_ESB_endtx();
+void NRF5_ESB_starttx()
+{
+	if (tx_retries > 0) {
+		// Prevent radio to write into TX memory while receiving
+		if (NRF_RADIO->PACKETPTR != (uint32_t)&tx_buffer) {
+			// Disable shorts
+			NRF_RADIO->SHORTS = 0;
+			// Disable radio
+			NRF_RADIO->TASKS_DISABLE = 1;
+		}
+
+		// Mark TX as unfinised
+		events_end_tx = false;
+
+		// Configure TX address to address at index NRF5_ESB_TX_ADDR
+		NRF_RADIO->TXADDRESS = NRF5_ESB_TX_ADDR;
+
+		// Enable TX when ready, Enable TX after disabling task
+		NRF_RADIO->SHORTS = NRF5_ESB_SHORTS_TX;
+
+		// reset timer
+		NRF_RESET_EVENT(NRF5_RADIO_TIMER->EVENTS_COMPARE[0]);
+		NRF_RESET_EVENT(NRF5_RADIO_TIMER->EVENTS_COMPARE[1]);
+		NRF5_RADIO_TIMER->TASKS_STOP = 1;
+		NRF5_RADIO_TIMER->TASKS_CLEAR = 1;
+		// Set retransmit time
+		NRF5_RADIO_TIMER->CC[1] = NRF5_ESB_ARD - NRF5_ESB_RAMP_UP_TIME;
+		// Set radio disable time to ACK_WAIT time
+		NRF5_RADIO_TIMER->CC[0] = NRF5_ESB_ACK_WAIT;
+
+		// Set PPI
+		NRF_PPI->CHENSET = NRF5_ESB_PPI_BITS;
+
+		// Disable Ready interrupt
+		NRF_RADIO->INTENCLR = RADIO_INTENSET_READY_Msk;
+
+		// Set buffer
+		NRF_RADIO->PACKETPTR = (uint32_t)&tx_buffer;
+
+		// Switch to TX
+		if (NRF_RADIO->STATE == RADIO_STATE_STATE_Disabled) {
+			NRF_RADIO->TASKS_TXEN = 1;
+		} else {
+			NRF_RADIO->TASKS_DISABLE = 1;
+		}
+	} else {
+		// finised TX
+		NRF5_ESB_endtx();
+	}
+	tx_retries--;
+}
+
+void NRF5_ESB_endtx()
+{
+	// Clear PPI
+	NRF_PPI->CHENCLR = NRF5_ESB_PPI_BITS;
+	// Enable Ready interrupt
+	NRF_RADIO->INTENSET = RADIO_INTENSET_READY_Msk;
+	// Stop Timer
+	NRF5_RADIO_TIMER->TASKS_STOP = 1;
+	// Shutdown timer
+	NRF5_RADIO_TIMER->TASKS_SHUTDOWN = 1;
+	// Mark TX as end
+	events_end_tx = true;
+	// Debug output
+#ifdef NRF5_ESB_DEBUG_INT_TX_END
+	NRF5_RADIO_DEBUG(PSTR("NRF5:INT:ENDTX\n"));
+#endif
+}
+
+static bool NRF5_ESB_sendMessage(uint8_t recipient, const void *buf, uint8_t len, const bool noACK)
+{
+	NRF5_RADIO_DEBUG(PSTR("NRF5:SND:TO=%d,LEN=%d,PID=%d,NOACK=%d\n"), recipient, len, tx_buffer.pid,
+	                 tx_buffer.noack); // send message
+	// Check if radio is initialized
+	if (NRF_RADIO->POWER == 0) {
+		NRF5_ESB_initialize();
+	}
+
+	// check length and truncate data
+	if (len > MAX_MESSAGE_LENGTH) {
+		len = MAX_MESSAGE_LENGTH;
+	}
+
+	// copy data to tx_buffer
+	memcpy(&tx_buffer.data[0], buf, len);
+
+	// build metadata
+	tx_buffer.len = len;
+	tx_buffer.noack = noACK;
```

Comment:
The RF24 driver sets the NOACK flag either for broadcasted messages or if the noACK parameter is set (used for passive node mode) - I suggest this change:

```tx_buffer.noack = noACK || recipient==BROADCAST_ADDRESS;```